_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

```diff
@@ -0,0 +1,281 @@
+<?php
+/*
+ * Service API Keys: Exposes 3rd party api keys that are used on a site.
+ *
+ * [
+ *   { # Availabilty Object. See schema for more detail.
+ *      code:            (string) Displays success if the oppration was successfully executed and an error code if it was not
+ *      service:         (string) The name of the service in question
+ *      service_api_key: (string) The API key used by the service empty if one is not set yet
+ *      message:         (string) User friendly message
+ *   },
+ *   ...
+ * ]
+ *
+ * @since 6.9
+ */
+class WPCOM_REST_API_V2_Endpoint_Service_API_Keys extends WP_REST_Controller {
+
+	function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'service-api-keys';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	public function register_routes() {
+		register_rest_route(
+			'wpcom/v2',
+			'/service-api-keys/(?P<service>[a-z\-_]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( __CLASS__, 'get_service_api_key' ), // todo move the methods to this class..
```

Comment:
Still a `// todo`?  Looks like you've moved the method to this class already :+1:

Reply:
(There's a few more such `// todo`s below)